URL: <https://sourceware.org/ml/gdb-patches/2012-02/msg00073.html>
Message-ID: <m3mx8uk5wa.fsf@fleche.redhat.com>

  From: Tom Tromey <tromey at redhat dot com>
  To: gdb-patches at sourceware dot org
  Subject: FYI: fix possible crash in find_charset_names
  Date: Tue, 07 Feb 2012 08:41:25 -0700
  
  We got a report in Red Hat bugzilla about a crash in find_charset_names:

  https://bugzilla.redhat.com/show_bug.cgi?id=786091

  I don't have a test case but the fix is obvious enough to go in without
  one.

  Tom

commit 1d6b2d2b4a864c478fb6a46e15a29859641fe788
Author: Tom Tromey <tromey@redhat.com>
Date:   Tue Feb 7 15:42:39 2012 +0000

    	* charset.c (find_charset_names): Check 'in' against NULL.

Index: gdb-7.2/gdb/charset.c
===================================================================
--- gdb-7.2.orig/gdb/charset.c
+++ gdb-7.2/gdb/charset.c
@@ -809,7 +809,7 @@ find_charset_names (void)
 	 parse the glibc and libiconv formats; feel free to add others
 	 as needed.  */
 
-      while (!feof (in))
+      while (in != NULL && !feof (in))
 	{
 	  /* The size of buf is chosen arbitrarily.  */
 	  char buf[1024];
